src/shogun/lib/DynamicObjectArray.h

```diff
@@ -32,13 +33,9 @@ class CDynamicObjectArray : public CSGObject
 {
 	public:
 		/** default constructor */
-		CDynamicObjectArray()
-		: CSGObject(), m_array()
+		CDynamicObjectArray() : CSGObject()
 		{
-			dim1_size=1;
-			dim2_size=1;
-			dim3_size=1;
-
+			m_array = std::vector<CSGObject*>();
```

Comment:
you need to initialize `dim1_size` `dim2_size` and `dim3_size`, otherwise it'll contain some random values and that's why the `SGObject.equals_DynamicObjectArray_equal` test will fail as those randoms are not the same. :)
so please leave there those initialisation values.